### PR TITLE
Replace invoke with nox

### PR DESCRIPTION
**Pull request by pcattori**
# ↪️ Pull Request
This PR replaces our previous task runner, `invoke`, with `nox`.

[nox](https://nox.thea.codes/en/stable/) allows for [testing multiple Python versions locally](https://nox.thea.codes/en/stable/tutorial.html#testing-against-different-and-multiple-pythons) as well as [passing arguments to underlying tools](https://nox.thea.codes/en/stable/config.html#passing-arguments-into-sessions) like `pytest`.

Also, linting is now being checked in `tests/` more broadly, so I fixed linting issues there (mostly removing `f` in front of strings that don't use formatting).

*For reviewers*: this PR can be reviewed commit-by-commit.

## 💻 Examples

```sh
alias prn='poetry run nox'
prn # run all checks
prn --list # see all possible sessions
# prn -s <session> to run specific session e.g.:
prn -s test # runs tests against all python versions
prn -s lint test-3.6 # runs tests only against Python 3.6
prn -s lint
```

See new contributor docs included in these changes.

## ✔️ PR Todo

- [n/a] Added/updated testing for this change
- [n/a] Included links to related issues/PRs
- [x] Update relevant [docs](https://github.com/Datatamer/tamr-client/tree/master/docs) + docstrings
- [n/a] Update the [CHANGELOG](https://github.com/Datatamer/tamr-client/blob/master/CHANGELOG.md) under the current `-dev` version:
  - Add changelog entries under any that apply: **BREAKING CHANGES**, **NEW FEATURES**, **BUG FIXES**.
  - Changelog entry format: `[#<issue number>](<link to issue>) <change description>`

### CODE REVIEW

**Comment by pcattori** [comment]
When we merge this, we need to remember to add the "Format" check as a required check for the repo.

File: .python-version [hunk 1]
```diff
@@ -0,0 +1,3 @@
+3.6.9
+3.7.7
+3.8.2
```
Comment (skalish) [inline .python-version:3]: It looks like these are almost the most recent minor versions of `3.6`, `3.7`, and `3.8`, but I see releases for `3.6.10` and `3.8.3`.  What's the reasoning behind choosing these specific versions, and is the idea to increment them as new minor versions are released?

File: docs/contributor-guide/run-and-build.md [hunk 2]
```diff
@@ -1,58 +1,78 @@
...
-pri test # with alias
-poetry run invoke test # without alias
+prn -s test # with alias
+poetry run nox -s test # without alias
 ```
 
-To run specific tests, see [these pytest docs](https://docs.pytest.org/en/latest/usage.html#specifying-tests-selecting-tests) and run `pytest` explicitly:
+---
+
+To run tests for a specifc Python version e.g. 3.6:
```
Comment (skalish) [inline docs/contributor-guide/run-and-build.md:61]: typo:
```suggestion
To run tests for a specific Python version e.g. 3.6:
```

File: .python-version (hunk 1 above)
Comment (pcattori) [reply .python-version:3]: I chose these based on what is currently available via `pyenv`. You're right about `3.6.10`, we can update to that, but `3.8.3` is not available via `pyenv` yet

**Pull request merged**